src/Bicep.Cli/Commands/FormatCommand.cs

```diff
@@ -54,50 +27,56 @@ public int Run(FormatArguments args)
+        var options = GetPrettyPrinterOptions(inputUri, args);
+        var context = PrettyPrinterV2Context.Create(options, parser.LexingErrorLookup, parser.ParsingErrorLookup);
 
-        if (featureProvider.PrettyPrintingEnabled)
-        {
-            var v2Options = this.configurationManager.GetConfiguration(inputUri).Formatting.Data;
-            var printerV2Context = PrettyPrinterV2Context.Create(v2Options, parser.LexingErrorLookup, parser.ParsingErrorLookup);
-
-            if (args.OutputToStdOut)
-            {
-                PrettyPrinterV2.PrintTo(io.Output, program, printerV2Context);
-                io.Output.Flush();
-            }
-            else
-            {
-                var outputPath = PathHelper.ResolveDefaultOutputPath(inputUri.LocalPath, args.OutputDir, args.OutputFile, path => path);
-                using var writer = new StreamWriter(outputPath);
-
-                PrettyPrinterV2.PrintTo(writer, program, printerV2Context);
-            }
-
-            return 0;
-        }
-
-        var options = new PrettyPrintOptions(
-            args.Newline ?? NewlineOption.Auto,
-            args.IndentKind ?? IndentKindOption.Space,
-            args.IndentSize ?? 2,
-            args.InsertFinalNewline ?? false
-        );
-
-        var output = PrettyPrinter.PrintProgram(program, options, parser.LexingErrorLookup, parser.ParsingErrorLookup);
         if (args.OutputToStdOut)
         {
-            io.Output.Write(output);
+            PrettyPrinterV2.PrintTo(io.Output, program, context);
             io.Output.Flush();
         }
         else
         {
-            static string DefaultOutputPath(string path) => path;
-            var outputPath = PathHelper.ResolveDefaultOutputPath(inputUri.LocalPath, args.OutputDir, args.OutputFile, DefaultOutputPath);
+            var outputPath = PathHelper.ResolveDefaultOutputPath(inputUri.LocalPath, args.OutputDir, args.OutputFile, path => path, fileSystem);
+            using var fileStream = fileSystem.File.Open(outputPath, FileMode.Create, FileAccess.Write, FileShare.Read);
+            using var writer = new StreamWriter(fileStream);
 
-            File.WriteAllText(outputPath, output);
+
+            PrettyPrinterV2.PrintTo(writer, program, context);
         }
 
         return 0;
     }
+
+    private PrettyPrinterV2Options GetPrettyPrinterOptions(Uri inputUri, FormatArguments args)
+    {
+        var options = configurationManager.GetConfiguration(inputUri).Formatting.Data;
+
+        if (args.NewlineKind is not null)
+        {
+            options = options with { NewlineKind = args.NewlineKind.Value };
+        }
+
+        if (args.IndentKind is not null)
+        {
+            options = options with { IndentKind = args.IndentKind.Value };
+        }
+
+        if (args.IndentSize is not null)
+        {
+            options = options with { IndentSize = args.IndentSize.Value };
+        }
+
+        if (args.Width is not null)
```

Comment:
Did you make a decision on what the default width should be?

Reply:
Not yet, but I'm leaning towards `100` as an optimal value - `80` feels a bit constrained, while `120` might be too wide for those without high-resolution screens. I'll arrange for a vote during our upcoming Bicep community call to decide and will update the code accordingly.